src/plugins/workspace/server/saved_objects/workspace_id_consumer_wrapper.ts

```diff
@@ -61,6 +62,52 @@ export class WorkspaceIdConsumerWrapper {
     return type === UI_SETTINGS_SAVED_OBJECTS_TYPE;
   }
 
+  private async checkWorkspacesExist(
+    workspaces: SavedObject['workspaces'] | null,
+    wrapperOptions: SavedObjectsClientWrapperOptions
+  ) {
+    if (workspaces?.length) {
+      let invalidWorkspaces: string[] = [];
+      // If only has one workspace, we should use get to optimize performance
+      if (workspaces.length === 1) {
+        const workspaceGet = await this.workspaceClient.get(
+          { request: wrapperOptions.request },
+          workspaces[0]
+        );
+        if (!workspaceGet.success) {
+          invalidWorkspaces = [workspaces[0]];
+        }
+      } else {
+        const workspaceList = await this.workspaceClient.list(
```

Comment:
I'm good with the current implementation, but perhaps we should introduce a new `workspaceClient.bulkGet` API, can we add a TODO item if it makes sense? @yubonluo @SuZhou-Joe